### PR TITLE
Feature/bap 33 allow scrolling of virtualized tables

### PR DESCRIPTION
**Pull request by afomina-quantori**
This branch is based on BAP-2, so we should integrate first this pull request:
https://github.com/quantori/Behavioral.Automation/pull/15

To review this pull request, see only changes in TableBinding file.

### CODE REVIEW

**Review by SergeyRuzin — DISMISSED** [review]

**Review by mikhail1aleksandrov1 — APPROVED** [review]

File: src/Behavioral.Automation/Bindings/ListBinding.cs [hunk 1]
```diff
@@ -24,8 +24,17 @@ public void CheckListContainsItems(IListWrapper list, string behavior, Table tab
             var expectedListValues = ListServices.TableToRowsList(table);
 
             bool exactOrder = behavior.Contains("contain in exact order");
-            bool check = list.ListValues.ContainsValues(expectedListValues, exactOrder);
-            check.Should().Be(!behavior.Contains("not contain"));
+
+            bool checkResult;
+            if (behavior.StartsWith("not"))
+            {
+                checkResult = list.ListValues.DoesntContainValues(expectedListValues);
```
Comment (afomina-quantori) [inline src/Behavioral.Automation/Bindings/ListBinding.cs:31]: we should check that list doesn't contain all listed values, so previous approach wasn't correct

File: src/Behavioral.Automation/Bindings/TableBinding.cs [hunk 2]
```diff
@@ -23,12 +24,23 @@ public void CheckRowsCount([NotNull] ITableWrapper element, int count)
         [Then("(.*?) should (contain|not contain) the following rows:")]
         public void CheckTableContainsRows(ITableWrapper gridRows, string behavior, Table table)
         {
-            var tableList = ListServices.TableToCellsList(table);
+            var expectedValues = ListServices.TableToCellsList(table);
+
             Assert.ShouldBecome(() => gridRows.Stale, false, $"{gridRows.Caption} is stale");
-            Assert.ShouldBecome(() => gridRows.Text.Contains(tableList.FirstOrDefault()), true, $"{gridRows.Caption} text is \"{gridRows.Text}\"");
```
Comment (afomina-quantori) [inline src/Behavioral.Automation/Bindings/TableBinding.cs:28]: sometimes we have tables without caption, so this check can't be done inside this binding.

File: src/Behavioral.Automation/Bindings/TableBinding.cs [hunk 3]
```diff
@@ -23,12 +24,23 @@ public void CheckRowsCount([NotNull] ITableWrapper element, int count)
...
-            var gridRowsCellsText = gridRows.CellsText.ToList();
-            Assert.ShouldBecome(() => ListServices.CompareTwoLists(gridRows.CellsText.ToList(), tableList), !behavior.Contains("not"), 
-                $"{gridRows.Caption} is {gridRowsCellsText.Aggregate((x, y) => $"{x}, {y}")}");
+
+            bool inversion = behavior.StartsWith("not");
+            if (inversion)
+            {
+                Assert.ShouldBecome(() => gridRows.CellsText.DoesntContainValues(expectedValues),
```
Comment (afomina-quantori) [inline src/Behavioral.Automation/Bindings/TableBinding.cs:34]: we should check that table doesn't contain all listed values

File: src/Behavioral.Automation/Services/ListServices.cs [hunk 4]
```diff
@@ -100,7 +101,7 @@ public static bool ContainsValues(this IEnumerable<string> actualCollection, Lis
             {
                 if (index > maxIndexForValuesList)
                 {
-                    break;
```
Comment (afomina-quantori) [inline src/Behavioral.Automation/Services/ListServices.cs:103]: if actual collection has more values than expected collection, it is ok, and check result should be = true

**Review by PavelAAlexeevQ — APPROVED** [review]

**Review by mikhail1aleksandrov1 — APPROVED** [review]

**Review by SergeyRuzin — APPROVED** [review]

**Pull request merged**